Shared/Samples/Create load report/CreateLoadReportView.Model.swift

```diff
@@ -0,0 +1,299 @@
+        
+        /// The phase summaries in the load report.
+        @Published private(set) var summaries = [CodedValue: PhaseSummary]()
+        
+        /// The number formatter for phase summaries.
+        private let numberFormatter: NumberFormatter = {
+            let formatter = NumberFormatter()
+            formatter.numberStyle = .decimal
+            formatter.usesGroupingSeparator = true
+            return formatter
+        }()
+        
+        /// A Boolean value indicating if the run button is enabled.
+        @Published private(set) var runEnabled = false
+        
+        /// A Boolean value indicating if the reset button is enabled.
+        @Published private(set) var resetEnabled = false
+        
+        /// The status text to display to the user.
+        @Published private(set) var statusText: String?
+        
+        /// A Boolean value indicating if the sample is authenticated.
+        private var isAuthenticated: Bool {
+            !ArcGISEnvironment.authenticationManager.arcGISCredentialStore.credentials.isEmpty
+        }
+        
+        // MARK: Methods
+        
+        /// Performs important tasks including adding credentials, loading and adding operational layers.
+        func setup() async throws {
+            try await ArcGISEnvironment.authenticationManager.arcGISCredentialStore.add(.publicSample)
+            try await loadUtilityNetwork()
+        }
+        
+        /// Load the utility network.
+        private func loadUtilityNetwork() async throws {
+            statusText = "Loading utility network…"
+            try? await utilityNetwork.load()
+            statusText = nil
+            
+            guard let startingLocation = makeStartingLocation(),
+                  // Get the base condition and trace configuration from a default tier.
+                  let traceConfiguration = getTraceConfiguration() else { return }
+            
+            // Set the default expression.
+            initialExpression = traceConfiguration.traversability?.barriers as? UtilityTraceConditionalExpression
+            
+            // Create downstream trace parameters with elements and function outputs.
+            let traceParameters = UtilityTraceParameters(traceType: .downstream, startingLocations: [startingLocation])
+            traceParameters.addResultTypes([.elements, .functionOutputs])
+            
+            // The service category for counting total customers.
+            if let serviceCategory = utilityNetwork.definition?.categories.first(where: { $0.name == "ServicePoint" }),
+               // The load attribute for counting total load.
+               let loadAttribute = utilityNetwork.definition?.networkAttributes.first(where: { $0.name == "Service Load" }),
+               // The phase attribute for getting total phase current load.
+               let phasesNetworkAttribute = utilityNetwork.definition?.networkAttributes.first(where: { $0.name == "Phases Current" }) {
```

Comment:
`utilityNetwork.definition` is conditionally unwrapped three times here, so it may be a good idea to unwrap it before calling the `first`s.